### PR TITLE
Add new parameter with settable option: cmakeargs

### PR DESCRIPTION
**Pull request by krytarowski**
This will allow to pass user-defined CMake arguments.

While there, there was need to alter loop type to handle options passed to
arguments. As a bonus, new loop should be POSIX confirming.

Improve curl dependency usage using standard CMake module to detect it.
This was required with pkgsrc on NetBSD.

### CODE REVIEW

**Comment by krytarowski** [comment]
@stephentoub PTAL

**Comment by krytarowski** [comment]
cmakeargs is modeled after option in CoreCLR

**Comment by stephentoub** [comment]
cc: @eerhardt, @nguerrera, @kapilash

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please

File: src/Native/System.Net.Http.Native/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,9 @@
 project(System.Net.Http.Native)
 
-find_library(CURL NAMES curl)
-if(CURL STREQUAL CURL-NOTFOUND)
-    message(SEND_ERROR "!!! Cannot find libcurl and System.Net.Http.Native cannot build without it. Try installing libcurl4-openssl-dev (or the appropriate package for your platform) !!!")
-    return()
-endif()
+find_package(CURL)
```
Comment (nguerrera) [inline src/Native/System.Net.Http.Native/CMakeLists.txt]: The changes to this file are not described in the PR or commit. What is the issue?

File: src/Native/System.Net.Http.Native/CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
+if(NOT CURL_FOUND)
+    message(FATAL_ERROR "!!! Cannot find libcurl and System.Net.Http.Native cannot build without it. Try installing libcurl4-openssl-dev (or the appropriate package for your platform) !!!")
+endif(NOT CURL_FOUND)
```
Comment (nguerrera) [inline src/Native/System.Net.Http.Native/CMakeLists.txt]: The curl issue is separate from the rest of the PR, right? Could you split the commits? You can keep the 1 PR.

(Disregard earlier deleted comment about it not being in the description, I just missed it.)

**Comment by nguerrera** [comment]
LGTM

File: src/Native/System.Net.Http.Native/CMakeLists.txt (hunk 2 above)
Comment (krytarowski) [inline src/Native/System.Net.Http.Native/CMakeLists.txt]: There are 2 issues under umbrella of CMake usage on NetBSD (with pkgsrc).

Is there need to split it into 2 commits? Or leave it as it is.

Comment (krytarowski) [inline src/Native/System.Net.Http.Native/CMakeLists.txt]: Done.

File: build.sh [hunk 3]
```diff
@@ -247,6 +248,7 @@ esac
 __BuildOS=$__HostOS
 __BuildType=Debug
 __CMakeArgs=DEBUG
+__cmakeextraargs=""
```
Comment (eerhardt) [inline build.sh]: (nit) Can this be cased like the existing variables?  (Pascal casing).

**Comment by eerhardt** [comment]
Two minor comments, otherwise this change looks good to me.  Thanks for the contribution!

File: build.sh (hunk 3 above)
Comment (krytarowski) [inline build.sh]: I will prepare the change.

**Pull request merged**